src/vcModals.cpp

```diff
@@ -619,6 +619,83 @@ void vcModals_DrawProfile(vcState* pProgramState)
   }
 }
 
+void vcModals_DrawChangePassword(vcState *pProgramState)
+{
+  const char *profile = vcString::Get("modalChangePasswordTitle");
+
+  if (pProgramState->openModals & (1 << vcMT_ChangePassword))
+    ImGui::OpenPopup(profile);
+
+  float width = 390.f;
+  float height = 150.f;
+  float buttonWidth = 80.f;
+  ImGui::SetNextWindowSize(ImVec2(width, height), ImGuiCond_Always);
+
+  if (ImGui::BeginPopupModal(profile, nullptr, ImGuiWindowFlags_NoResize))
+  {
+    pProgramState->modalOpen = true;
+
+    if (pProgramState->sessionInfo.isOffline)
+    {
+      ImGui::TextUnformatted(vcString::Get("modalProfileOffline"));
+    }
+    else
+    {
+      ImGuiInputTextFlags passwordFieldFlags = ImGuiInputTextFlags_Password;
+
+      ImGui::InputText(vcString::Get("modalProfileCurrentPassword"), &pProgramState->changePassword.currentPassword[0], vcMaxPathLength, passwordFieldFlags);
+      ImGui::Separator();
+      ImGui::InputText(vcString::Get("modalProfileNewPassword"), &pProgramState->changePassword.newPassword[0], vcMaxPathLength, passwordFieldFlags);
+      ImGui::InputText(vcString::Get("modalProfileReEnterNewPassword"), &pProgramState->changePassword.newPasswordConfirm[0], vcMaxPathLength, passwordFieldFlags);
+    }
+
+    ImGui::Separator();
+    
+    if (ImGui::Button(vcString::Get("modalProfileConfirmNewPassword"), ImVec2(130.0f, 0)) || vcHotkey::IsPressed(vcB_Cancel))
+    {
+      udJSON changePasswordData;
+      changePasswordData.Set("username = \"%s\"", pProgramState->settings.loginInfo.username);
+      changePasswordData.Set("oldpassword = \"%s\"", pProgramState->changePassword.currentPassword);
+      changePasswordData.Set("password = \"%s\"", pProgramState->changePassword.newPassword);
+      changePasswordData.Set("passwordConfirm = \"%s\"", pProgramState->changePassword.newPasswordConfirm);
+
+      const char *pUpdatePasswordString = nullptr;
+      changePasswordData.Export(&pUpdatePasswordString, udJEO_JSON);
+
+      const char *pResult = nullptr;
+      vdkError result = vdkServerAPI_Query(pProgramState->pVDKContext, "v1/user/updatepassword", pUpdatePasswordString, &pResult);
+      if (result == vE_Success)
+      {
+        udJSON resultData;
+        resultData.Parse(pResult);
+
+        if (resultData.Get("success").AsBool())
+        {
+          ImGui::CloseCurrentPopup();
+          memset(pProgramState->changePassword.currentPassword, 0, sizeof(pProgramState->changePassword.currentPassword));
+          memset(pProgramState->changePassword.newPassword, 0, sizeof(pProgramState->changePassword.newPassword));
+          memset(pProgramState->changePassword.newPasswordConfirm, 0, sizeof(pProgramState->changePassword.newPasswordConfirm));
+          memset(pProgramState->changePassword.message, 0, sizeof(pProgramState->changePassword.message));
+        }
+        else
+        {
+          udStrcpy(pProgramState->changePassword.message, resultData.Get("message").AsString());
```

Comment:
These messages are going to need to be translated. I would recommend using `udStrEqual` to determine if they're the string you expect, and then substitute in a translation key. When it's none of the known strings, display unknown error.

Reply:
so the string result that comes out of vdkServerAPI_Query will always be the english text? If so, I can have something that does a strcmp with the expected text, and if they match, just use the text from the language enAU.json file

Reply:
This is fixed in latest push. not sure why there's no "outdated" tag on it.